### DISCUSSION TITLE
Idea: Drag a signal to a node

### DISCUSSION THREAD

**Discussion by theraot** [godotengine/godot-proposals#7552] [Editor]
I would like to be able to drag a signal from the Signals tab of Node dock, to a Node in the Scene dock in the editor※ and have it open the dialog to connect the signal with that Node selected.

※: Or even in the main 2D or 3D area, which I understand won't be perfect due to overlapping Nodes.

I would find this useful for scenes with lots of Nodes, so I can then proceed to set a method name or pick an existing one (I should still be able to change the Node, for example if I dragged the signal to a collider but I want the physics body).

This could save time and cognitive load, in particular in scenes with lots of nodes, where connections are not necessarily between nearby nodes.

*My use case is wiring a puzzle in 3D.*

---

Other related ideas:

- Highlight connected nodes when a signal is selected, or the connected node when a connection is selected, so I can quickly check if it is connected to the correct Node.
- When I right click on a signal connection I get "Edit..", "Go To Method" and "Disconnect", I would like "Select Node".

Addendum: When connecting to an existing method, I might not want to switch to the script automatically.

**Comment by timothyqiu** [comment]
There're already a few related proposals:

* #1759
* #3953
* #6347
